connect-inject/container_init_test.go

```diff
@@ -91,6 +91,27 @@ func TestHandlerContainerInit(t *testing.T) {
 			"",
 			`datacenter`,
 		},
+		{
+			"Prepared Query specified",
+			func(pod *corev1.Pod) *corev1.Pod {
+				pod.Annotations[annotationService] = "web"
+				pod.Annotations[annotationUpstreams] = "prepared_query:handle:1234"
+				return pod
+			},
+			`destination_type = "prepared_query"`,
+			"",
+		},
+
+		{
+			"Prepared Query specified",
```

Comment:
As with the other test case, mentioning in the test title that this one checks that the `destination_name` is set correctly from a query annotation would help make its purpose more explicit.